content/docs/1.2.4/volumes-and-nodes/volume-size.md

```diff
@@ -19,29 +19,90 @@ In this section, you'll have a better understanding of concepts related to volum
-    - The `actual size` is still 2Gi.
-{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-2.png" >}}
-4. Write 4Gi data (data#2) in the volume mount, then take one more snapshot (snapshot#2). See Figure 3 of the illustration.
-    - Now the `actual size` is 6Gi, which is greater than the nominal `size`.
-    - There is overlapping between 2 snapshots in block level (See the 2 snapshots in Figure 3), since the data#1 is marked as removed in snapshot#2 then the filesystem reuses the space.
-{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-3.png" >}}
-5. Delete snapshot#1 and wait for snapshot purge complete. See Figure 4 of the illustration.
+    - The `actual size` is still 8.25 Gi.
+
+7. Write 8 Gi data (data#2) in the volume mount, then take one more snapshot (snapshot#2). See Figure 5 of the illustration.
+    - Now the `actual size` is 16.3 Gi, which is greater than the volume nominal `size`.
+    - There is overlapping between 2 snapshots in block level (See the 2 snapshots in Figure 6), since the data#1 is marked as removed in snapshot#2 then the filesystem reuses the space.
+
+    > The volume head holds the latest data of the volume only, while each snapshot may store historical data as well as active data, which consumes at most size space. Therefore, the volume `actual size`, which is the size sum of the volume head and all snapshots, is possibly bigger than the size specified by users.
+    >
+    > Even if users will not take snapshots for volumes, there are operations like rebuilding, expansion, or backing up that would lead to system (hidden) snapshot creation. As a result, volume `actual size` being larger than size is unavoidable under some use cases.
+
+{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-illustration-fig5.png" >}}
+
+8. Delete snapshot#1 and wait for snapshot purge complete. See Figure 7 of the illustration.
     - Here Longhorn actually coalesces the snapshot#1 with the snapshot#2.
-    - For the overlapping part during the coalesce, the newer data (data#2) will be retained in the blocks. Then some historical data is removed and the volume gets shrunk (from 6.1Gi to 4.65Gi in the example).
-{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-example-4.png" >}}
+    - For the overlapping part during the coalesce, the newer data (data#2) will be retained in the blocks. Then some historical data is removed and the volume gets shrunk (from 16.3 Gi to 11.4 Gi in the example).
+
+{{< figure src="/img/screenshots/volumes-and-nodes/volume-size-illustration-fig6.png" >}}
+
+Here we summarize the issues we encountered in the example:
+1. Unused blocks not releasing issue
+    Longhorn does not support TRIM/UNMAP operations. Hence deleting files from filesystems will not lead to volume actual size decreasing/shrinking.
```

Comment:
Add an extra new line for formatting, otherwise, this will just immediately follow up with the above line.

```suggestion

    Longhorn does not support TRIM/UNMAP operations. Hence deleting files from filesystems will not lead to volume actual size decreasing/shrinking.
```